enum/enum.go

```diff
@@ -242,9 +245,7 @@ func (e *Enumeration) makeOutputSink() pipeline.SinkFunc {
 	})
 }
 
-func (e *Enumeration) filterTaskFunc() pipeline.TaskFunc {
-	filter := stringset.New()
-
+func (e *Enumeration) filterTaskFunc(filter *stringset.Set) pipeline.TaskFunc {
```

Comment:
The filter can be accessed from the `Enumeration` receiver instead of via the parameter